### PR TITLE
refactor(tailwind): update borderColor, fontSize, backgroundColor, fill config

### PR DESCRIPTION
**Pull request by jrowlingson**
Supports https://github.com/Esri/calcite-components/issues/1500 (https://github.com/Esri/calcite-components/issues/1501)

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -47,18 +55,18 @@ module.exports = {
...
+      "-1": [ "var(--calcite-font-size--1)", { lineHeight: '1rem' } ],    // 14px (0.875rem)
+      "0": [ "var(--calcite-font-size-0)", { lineHeight: '1.25rem' } ],   // 16px (1rem)
+      "1": [ "var(--calcite-font-size-1)", { lineHeight: '1.5rem' } ],    // 18px (1.125rem)
+      "2": [ "var(--calcite-font-size-2)", { lineHeight: '1.5rem' } ],    // 20px (1.25rem)
+      "3": [ "var(--calcite-font-size-3)", { lineHeight: '2rem' } ],      // 26px (1.625rem)
+      "4": [ "var(--calcite-font-size-4)", { lineHeight: '2.5rem' } ],    // 32px (2rem)
+      "5": [ "var(--calcite-font-size-5)", { lineHeight: '3rem' } ],      // 40px (2.5rem)
+      "6": [ "var(--calcite-font-size-6)", { lineHeight: '4rem' } ],      // 48px (3rem)
+      "7": [ "var(--calcite-font-size-7)", { lineHeight: '4rem' } ],      // 56px (3.5rem)
+      "8": [ "var(--calcite-font-size-8)", { lineHeight: '5rem' } ],      // 64px (4rem)
```
Comment (jrowlingson) [inline tailwind.config.js:69]: Including `line-height` inspired by the v2.x font size [utils](https://tailwindcss.com/docs/font-size). These are based on @bstifle 4px scale.

File: tailwind.config.js [hunk 2]
```diff
@@ -3,14 +3,23 @@ var flattenColorPalette = require('tailwindcss/lib/util/flattenColorPalette').de
...
+      },
+      "color-brand": theme("colors.brand"),
+      "color-brand-hover": theme("colors.brand-hover"),
+      "color-brand-press": theme("colors.brand-press"),
+      "color-info": theme("colors.info"),
+      "color-success": theme("colors.success"),
+      "color-warning": theme("colors.warning"),
+      "color-danger": theme("colors.danger"),
+      "color-danger-hover": theme("colors.danger-hover"),
+      "color-danger-press": theme("colors.danger-press"),
```
Comment (jrowlingson) [inline tailwind.config.js:21]: `theme()` can only be used with "top-level theme keys, not the keys inside of each section" [[1]](https://tailwindcss.com/docs/theme#referencing-other-values)

File: tailwind.config.js [hunk 3]
```diff
@@ -74,15 +95,22 @@ module.exports = {
...
     },
     textColor: theme => ({
-      color: {
-        "1": theme("colors.text.1"),
-        "2": theme("colors.text.2"),
-        "3": theme("colors.text.3"),
-        "link": theme("colors.text.link"),
-        "inverse": theme("colors.text.inverse")
-      }
+      color: theme("colors.text")
```
Comment (paulcpederson) [inline tailwind.config.js:98]: oh this is nice

**Comment by paulcpederson** [comment]
@jrowlingson is your branch up to date with `master`? I seem to remember fixing those contrast errors the build is throwing already...

**Comment by bstifle** [comment]
@jrowlingson 

https://github.com/Esri/calcite-components/blob/d288018b439a35e405b6c90e883d5acbd5fe5c4e/tailwind.config.js#L123


what is the "1-lg" "2-lg" "1-sm" "2-sm" naming coming from?

Is it possible to do shadow-1, shadow-1-hover, shadow-1-press  etc for level 2

**Comment by jrowlingson** [comment]
@bstifle I can submit a follow on to refactor the shadows to use the `-hover|press` naming.

> curious as to what this is too 0: "0 4px 8px -1px rgba(0, 0, 0, 0.08), 0 2px 4px -1px rgba(0, 0, 0, 0.04)"

I believe this was inherited from https://github.com/Esri/calcite-components/blame/master/src/assets/styles/_shadow.scss#L5. We could remove this is well if other are good with that.

**Comment by asangma** [comment]
@bstifle re: shadows
Would we want to update the [shadows in Base](https://github.com/Esri/calcite-base/blob/master/dist/variables.css#L22-L27)?

**Comment by bstifle** [comment]
i dont see why we wouldnt! @macandcheese any objecticons?

**Comment by julio8a** [comment]
Looping in @paulcpederson too as we're considering too, as we're considering not using the old base repository moving forward.

**Comment by paulcpederson** [comment]
> we're considering not using the old base repository moving forward.

Yeah I think the idea is to have "base" be just the tailwind config eventually.

**Comment by paulcpederson** [comment]
We could probably update base for now, though until we make that call for sure.

**Comment by jrowlingson** [comment]
Is this good for a merge? I've added the shadow update as a line item to #1500.

**Review by asangma — APPROVED** [review]
These updates look good to me.
Shadow stuff can be handled separately.

**Pull request merged**